src/components/Card.js

```diff
@@ -1,22 +1,50 @@
 export default class Card {
-  constructor({ data, handleImageClick }, cardSelector) {
+  constructor({ data, myId, handleLikeClick, handleDeleteClick, handleImageClick}, cardSelector) {
     this._data = data;
     this._name = data.name;
     this._link = data.link;
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
+    this._id = data._id;
+    this._myId = myId;
+    this._handleLikeClick = handleLikeClick;
+    this._handleDeleteClick = handleDeleteClick;
+    this._likes = data.likes;
+    this._ownerId = data.owner._id;
+  //  console.log(data._id)
+  //  console.log(myId)
+   console.log(data.owner._id)
   }
 
+  getId() {
+    return this._id;
+  }
+  isLiked() {
+    return this._likes.some((like) => like._id === this._myId);
+  }
+
+  setLikes(likes) {
+    this._likes = likes;
+    this._handleLikeIcon();
+    this.updateLikes();
+  }
+  updateLikes() {
+    this._likesAmount = this._cardElement.querySelector(".card__like-amount");
```

Comment:
Better to save the elements when initializing the card element and reuse them in other methods